### ISSUE TITLE
Can show_motd not replace ServerName custom message?

### ISSUE THREAD

**Issue by basuritashka** [alliedmodders/amxmodx#872]
Not sure if this is intended or a unexpected behavior but when calling ServerName message and editing it to display a custom one, when calling show_motd that custom message will get replaced with the original hostname, not sure if this should work like that but I think that should be addressed if possible. (Unless it's something client-side).

**Comment by SmileYzn** [comment]
motd function always will show hostname by default if string for title is empty i guess

**Comment by basuritashka** [comment]
> motd function always will show hostname by default if string for title is empty i guess

String empty or not it will get replaced.

**Comment by SmileYzn** [comment]
Did you have an test case code?

You calling ServerName and trying to display a motd?

**Comment by basuritashka** [comment]
If you set a custom ServerName hooking the message, once you open a mote, that value will get replaced with defaul servername, thats all.

**Comment by SmileYzn** [comment]
https://github.com/alliedmodders/amxmodx/blob/c86813697acf3a3b577ca35426053db8dd7f8902/amxmodx/util.cpp#L66

UTIL_ShowMOTD uses ServerName Message, look at the begin and end of function. Since MOTD messages aways display a serverName, what amxx team do is send ServerName message with motd title by show_motd param. And at the end send's the original hostname on ServerName message.

Now i have sure that you need to do something to avoid that on ServerName message hook.
Or just show a code example of how you hook in ServerName message.

**Comment by basuritashka** [comment]
There should be a way from the native itself to not hook ServerName it is quite annoying, literally you need a permanent task that set the custom ServerName and that will do the work, but thats not a great idea.

**Comment by basuritashka** [comment]
It is a simple thing what I do.

```register_message( get_user_msgid( "ServerName" ), "OnServerName" );```

```public OnServerName( )
{
	set_msg_arg_string( 1, "Hi" );
}
```

**Comment by SmileYzn** [comment]
ServerName message is fine, i think it is related wth this?

https://forums.alliedmods.net/showthread.php?t=114873

**Comment by basuritashka** [comment]
> ServerName message is fine, i think it is related wth this?
> 
> https://forums.alliedmods.net/showthread.php?t=114873
> 
> Sinte this change string itself, also you can try to compare with hostname cvar

Not sure if that will works, if show_motd gets executed, ServerName will restore his default value, nothing will change AFAIK.

**Comment by SmileYzn** [comment]
I guess when amxmodx call ServerName messages this is not reflected on registered message, so this is buggy. This is the real problem.

A solution is a pull request to call message

**Comment by basuritashka** [comment]
No, I don't think that will do the work, each call of show_motd will replace ServerName on client-side regardless of execution. This is a common unexpected behavior I guess, many friends have the same problem and are using or a permanent set_task or avoiding the usage of show_motd, my mod at least calls show_motd many times because there is a menu which explains concepts/rules of my mod.

**Comment by SmileYzn** [comment]
In each call to show_motd need to call ServerName message to avoid that, or you do in your code, or a future fix from amxx team will do in their code. Anyway i will try to make a pull request later to change this

**Comment by basuritashka** [comment]
Yeah, that may be a good solution, save ServerName changed value somewhere (like cache?) and then restoring it once show_motd gets displayed. 

@Arkshine any insight?
